sklearn/base.py

```diff
@@ -449,6 +449,25 @@ def fit_transform(self, X, y=None, **fit_params):
             return self.fit(X, y, **fit_params).transform(X)
 
 
+class DensityMixin(object):
+    """Mixin class for all density estimators in scikit-learn."""
+    _estimator_type = "DensityEstimator"
+
+    def score(self, X, y = None):
+        """Returns the log likelihood under the model on the given data X.
+
+        Parameters
+        ----------
+        X : array-like, shape = (n_samples, n_features)
+
+        Returns
+        -------
+        score: float
+            sum of log probability of samples
+        """
+        return np.sum(self.score_samples(X))
```

Comment:
I wonder if we should not use the "mean" log likelihood as the `score` method for the density models. Otherwise we will have a `score` function that is not [statistically consistent](https://en.wikipedia.org/wiki/Consistency_%28statistics%29): it will grow unbounded as you add more test samples which is a very unexpected behavior and is not consistent with the default score method for classifier (mean accuracy) and regressors (empirical r2).

In particular if the `score` function is used to evaluate overfitting, the training score and the test score would have to be evaluated on a training set and a test set with the same number of samples, otherwise the results cannot be compared.

My proposal would be to do use instead: `np.mean(self.score_samples(X))`. If we use that as the score we should make it clear that this is the mean log-likelihood instead of the total log likelihood.

What do people (@xuewei4d, @amueller, @jakevdp) think of that suggestion?